### PR TITLE
Added utility to compare 2 xla metrics reports.

### PR DESCRIPTION
**Pull request by taylanbil**
Usage and sample output:
```
$ python metrics_compare.py log-regression-0420.txt log-regression-0504.txt -c 1000 -p 5000 -t 10
Keys are exactly the same in reports.

--------------------------------------------------------------------------------

Changes sorted by pct -- Counters -- min threshold 10.0 -- max rows 1000

                                      KEY       Val1       Val2  PCT_CHANGE
           TransferToServerTime.Total_sec    14.1210   568.2095      3923.9
                       OutboundData.Count       1012       3188       215.0
               TransferToServerTime.Count       1012       3188       215.0
      TransferToServerTransformTime.Count       1012       3188       215.0
  TransferToServerTransformTime.Total_sec     0.9551     2.5844       170.6
              XRTAllocateFromTensor_Empty        268        526        96.3
                 DeviceLockWait.Total_sec   582.5619    42.2933       -92.7
                    CompileTime.Total_sec    48.4075    67.9889        40.5
             ReleaseDataHandlesTime.Count      16400      22795        39.0
         ReleaseDataHandlesTime.Total_sec    24.9688    34.1774        36.9
              InputOutputAliasCount.Total  1724.0000  2351.0000        36.4
         IrValueTensorToXlaData.Total_sec     2.9306     2.1857       -25.4
                        CompileTime.Count          7          8        14.3
              InputOutputAliasCount.Count          7          8        14.3
                     CreateCompileHandles          7          8        14.3
                          UncachedCompile          7          8        14.3

--------------------------------------------------------------------------------

Changes sorted by pct -- Percentiles -- min threshold 10.0 -- max rows 5000

                                    KEY     Val1     Val2  PCT_CHANGE
            TransferToServerTime.P1_sec   0.0010   0.0010         6.2
            TransferToServerTime.P5_sec   0.0011   0.0012         6.5
           TransferToServerTime.P10_sec   0.0012   0.0013         9.4
           TransferToServerTime.P20_sec   0.0013   0.0025       100.0
           TransferToServerTime.P50_sec   0.0015   0.2718     18479.2
           TransferToServerTime.P80_sec   0.0028   0.2726      9547.5
           TransferToServerTime.P90_sec   0.0032   0.2729      8433.0
           TransferToServerTime.P95_sec   0.0227   0.2722      1097.1
           TransferToServerTime.P99_sec   0.3595   0.2738       -23.8
          ReleaseDataHandlesTime.P1_sec   0.0005   0.0005        -7.5
          ReleaseDataHandlesTime.P5_sec   0.0006   0.0006        -3.1
         ReleaseDataHandlesTime.P10_sec   0.0007   0.0006        -5.1
         ReleaseDataHandlesTime.P20_sec   0.0008   0.0007        -5.7
         ReleaseDataHandlesTime.P50_sec   0.0010   0.0009        -2.9
         ReleaseDataHandlesTime.P80_sec   0.0012   0.0012         0.3
         ReleaseDataHandlesTime.P90_sec   0.0029   0.0026        -8.8
         ReleaseDataHandlesTime.P95_sec   0.0034   0.0045        32.3
         ReleaseDataHandlesTime.P99_sec   0.0044   0.0228       418.1
                     CompileTime.P1_sec   0.0571   0.0058       -89.8
                     CompileTime.P5_sec   0.0571   0.0058       -89.8
                    CompileTime.P10_sec   0.0571   0.0058       -89.8
                    CompileTime.P20_sec   0.0766   0.0744        -2.8
                    CompileTime.P50_sec   9.4492  10.7367        13.6
                    CompileTime.P80_sec  14.6641  17.4565        19.0
                    CompileTime.P90_sec  15.2243  21.0007        37.9
                    CompileTime.P95_sec  15.2243  21.0007        37.9
                    CompileTime.P99_sec  15.2243  21.0007        37.9
                  DeviceLockWait.P1_sec   0.0000   0.0000        -1.2
                  DeviceLockWait.P5_sec   0.0000   0.0000         0.8
                 DeviceLockWait.P10_sec   0.0000   0.0000        -2.5
                 DeviceLockWait.P20_sec   0.2298   0.0000      -100.0
                 DeviceLockWait.P50_sec   0.2386   0.0000      -100.0
                 DeviceLockWait.P80_sec   0.2393   0.0000      -100.0
                 DeviceLockWait.P90_sec   0.2401   0.0000      -100.0
                 DeviceLockWait.P95_sec   0.2728   0.2736         0.3
                 DeviceLockWait.P99_sec   0.2739   0.2740         0.0
   TransferToServerTransformTime.P1_sec   0.0001   0.0001        11.7
   TransferToServerTransformTime.P5_sec   0.0001   0.0001        15.6
  TransferToServerTransformTime.P10_sec   0.0001   0.0002        14.1
  TransferToServerTransformTime.P20_sec   0.0001   0.0002        15.3
  TransferToServerTransformTime.P50_sec   0.0002   0.0002         8.3
  TransferToServerTransformTime.P80_sec   0.0002   0.0002        -3.0
  TransferToServerTransformTime.P90_sec   0.0004   0.0002       -36.1
  TransferToServerTransformTime.P95_sec   0.0056   0.0003       -95.4
  TransferToServerTransformTime.P99_sec   0.0114   0.0009       -92.5
                     OutboundData.P1_mb   0.0000   0.0000         0.0
                     OutboundData.P5_mb   0.0000   0.0000         0.0
                    OutboundData.P10_mb   0.0000   0.0000         0.0
                    OutboundData.P20_mb   0.0000   0.0000         0.0
                    OutboundData.P50_mb   0.0000   0.0000         0.0
                    OutboundData.P80_mb   0.0000   0.0000         0.0
                    OutboundData.P90_mb   0.0045   0.0000       -99.8
                    OutboundData.P95_mb   4.5000   0.0000      -100.0
                    OutboundData.P99_mb   8.0200   0.7695       -90.4
          IrValueTensorToXlaData.P1_sec   0.0009   0.0011        26.6
          IrValueTensorToXlaData.P5_sec   0.0010   0.0012        19.8
         IrValueTensorToXlaData.P10_sec   0.0011   0.0013        22.6
         IrValueTensorToXlaData.P20_sec   0.0012   0.0015        24.6
         IrValueTensorToXlaData.P50_sec   0.0026   0.0020       -22.2
         IrValueTensorToXlaData.P80_sec   0.0174   0.0171        -1.7
         IrValueTensorToXlaData.P90_sec   0.0213   0.0228         7.3
         IrValueTensorToXlaData.P95_sec   0.0242   0.0262         8.1
         IrValueTensorToXlaData.P99_sec   0.3358   0.3625         7.9
          TransferFromServerTime.P1_sec   0.0009   0.0010         4.1
          TransferFromServerTime.P5_sec   0.0010   0.0010         2.7
         TransferFromServerTime.P10_sec   0.0010   0.0011         5.6
         TransferFromServerTime.P20_sec   0.0010   0.0011         7.0
         TransferFromServerTime.P50_sec   0.0011   0.0013         9.2
         TransferFromServerTime.P80_sec   0.0013   0.0015         9.9
         TransferFromServerTime.P90_sec   0.0015   0.0026        76.2
         TransferFromServerTime.P95_sec   0.0028   0.0028         1.6
```

I sorted the percentiles by group-priority, which is the avg percentile change over all percentiles per metric.

### CODE REVIEW

**Comment by taylanbil** [comment]
@dlibenzi I added pretty much all you asked except for humanization. Could we leave that for another pr?

**Comment by dlibenzi** [comment]
> @dlibenzi I added pretty much all you asked except for humanization. Could we leave that for another pr?

Why?
Let's do it here.
Remove _sec _mb ...

There is a package for humanization 😄 

https://pypi.org/project/humanize/

**Comment by taylanbil** [comment]
Oh for a second I thought I would have to implement that myself..
*facepalm*. Well that makes it easy, I ll add humanization tmrw. Thanks.

**Comment by taylanbil** [comment]
Here's the sample output now:
```
(torch-xla-nightly) pytorch-xla-europe➜  20200424-robertatutregression  ᐅ  python metrics_compare.py log-regression-0420.txt log-regression-0504.txt -c 10 -p 50 -t 20
Keys are exactly the same in reports.

--------------------------------------------------------------------------------

Changes sorted by pct -- Counters -- min threshold 20.0 -- max rows 10

                                  KEY             Val1       Val2  PCT_CHANGE
           TransferToServerTime.Total           14 sec  9 minutes      3923.9
                   OutboundData.Count             1012       3188       215.0
           TransferToServerTime.Count             1012       3188       215.0
  TransferToServerTransformTime.Count             1012       3188       215.0
  TransferToServerTransformTime.Total  955054 microsec      2 sec       170.6
          XRTAllocateFromTensor_Empty              268        526        96.3
                 DeviceLockWait.Total        9 minutes     42 sec       -92.7
                    CompileTime.Total           48 sec   a minute        40.5
         ReleaseDataHandlesTime.Count            16400      22795        39.0
         ReleaseDataHandlesTime.Total           24 sec     34 sec        36.9

--------------------------------------------------------------------------------

Changes sorted by pct -- Percentiles -- min threshold 20.0 -- max rows 50

                               KEY             Val1             Val2  PCT_CHANGE
           TransferToServerTime.P1     987 microsec    1047 microsec         6.2
                                P5    1090 microsec    1162 microsec         6.5
                               P10    1152 microsec    1261 microsec         9.4
                               P20    1250 microsec    2501 microsec       100.0
                               P50    1463 microsec  271782 microsec     18479.2
                               P80    2826 microsec  272626 microsec      9547.5
                               P90    3198 microsec  272893 microsec      8433.0
                               P95   22735 microsec  272151 microsec      1097.1
                               P99  359479 microsec  273777 microsec       -23.8
         ReleaseDataHandlesTime.P1     539 microsec     499 microsec        -7.5
                                P5     597 microsec     578 microsec        -3.1
                               P10     662 microsec     628 microsec        -5.1
                               P20     773 microsec     729 microsec        -5.7
                               P50     964 microsec     936 microsec        -2.9
                               P80    1230 microsec    1233 microsec         0.3
                               P90    2878 microsec    2625 microsec        -8.8
                               P95    3407 microsec    4507 microsec        32.3
                               P99    4404 microsec   22818 microsec       418.1
                    CompileTime.P1   57091 microsec    5822 microsec       -89.8
                                P5   57091 microsec    5822 microsec       -89.8
                               P10   57091 microsec    5822 microsec       -89.8
                               P20   76607 microsec   74444 microsec        -2.8
                               P50            9 sec           10 sec        13.6
                               P80           14 sec           17 sec        19.0
                               P90           15 sec           21 sec        37.9
                               P95           15 sec           21 sec        37.9
                               P99           15 sec           21 sec        37.9
                 DeviceLockWait.P1       6 microsec       6 microsec        -1.2
                                P5       6 microsec       7 microsec         0.8
                               P10       7 microsec       7 microsec        -2.5
                               P20  229792 microsec       7 microsec      -100.0
                               P50  238636 microsec       8 microsec      -100.0
                               P80  239345 microsec       8 microsec      -100.0
                               P90  240087 microsec       9 microsec      -100.0
                               P95  272783 microsec  273607 microsec         0.3
                               P99  273946 microsec  274021 microsec         0.0
  TransferToServerTransformTime.P1     113 microsec     127 microsec        11.7
                                P5     126 microsec     145 microsec        15.6
                               P10     137 microsec     156 microsec        14.1
                               P20     147 microsec     170 microsec        15.3
                               P50     175 microsec     189 microsec         8.3
                               P80     221 microsec     214 microsec        -3.0
                               P90     372 microsec     237 microsec       -36.1
                               P95    5636 microsec     258 microsec       -95.4
                               P99   11351 microsec     855 microsec       -92.5
                   OutboundData.P1          0 Bytes          0 Bytes         0.0
                                P5          0 Bytes          0 Bytes         0.0
                               P10          0 Bytes          0 Bytes         0.0
                               P20          0 Bytes          0 Bytes         0.0
                               P50          0 Bytes          0 Bytes         0.0
```

File: scripts/metrics_compare.py [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+
+def main(args):
+  report1 = extract_report(args.filepath1, args.skip_1)
+  report2 = extract_report(args.filepath2, args.skip_2)
+  report1 = mcu.parse_metrics_report(report1)
+  report2 = mcu.parse_metrics_report(report2)
+  print_comparison_summary(args, report1, report2)
+
+
+if __name__ == "__main__":
```
Comment (dlibenzi) [inline scripts/metrics_compare.py:201]: Use single quotes for strings is the rule we follow.

**Comment by dlibenzi** [comment]
Remember to Squash!

**Pull request merged**